pkg/steps/multi_stage.go

```diff
@@ -334,6 +342,37 @@ func (s *multiStageTestStep) createCredentials() error {
 	return nil
 }
 
+func (s *multiStageTestStep) createCommandConfigMaps() error {
+	log.Printf("Creating multi-stage test command configmaps for %q", s.name)
+	toCreate := map[string]*coreapi.ConfigMap{}
+	for _, step := range append(s.pre, append(s.test, s.post...)...) {
+		name := commandConfigMapForStep(s.name, step.As)
+		toCreate[name] = &coreapi.ConfigMap{
+			ObjectMeta: meta.ObjectMeta{
+				Name:      name,
+				Namespace: s.jobSpec.Namespace(),
+			},
+			Data: map[string]string{
+				CommandScriptFilename: step.Commands,
+			},
+		}
+	}
+	for name := range toCreate {
+		// delete old command configmaps if they exist
+		if err := s.client.Delete(context.TODO(), toCreate[name]); err != nil && !kerrors.IsNotFound(err) {
+			return fmt.Errorf("could not delete command configmap %s: %w", toCreate[name].Name, err)
+		}
+		if err := s.client.Create(context.TODO(), toCreate[name]); err != nil {
+			return fmt.Errorf("could not create command configmap: %w", err)
+		}
+	}
+	return nil
+}
+
+func commandConfigMapForStep(testName, stepName string) string {
+	return fmt.Sprintf("%s-%s-command-script", testName, stepName)
```

Comment:
Will this ever be longer than the maximum length of a k8s object name?

Reply:
Quite unlikely, but I assume it's possible. I could change the name to `testName-stepName`, which is the name used for the pod as well, to slightly shorten it.